### PR TITLE
[RA-360][SST-20272] Bump Rails to 6.1.7

### PR DESCRIPTION
**Pull request by amaury-sie**
## Why?
[RA-360](https://tractionguest.atlassian.net/browse/RA-360)
[SST-20272](https://tractionguest.atlassian.net/browse/SST-20272)

What is the problem this PR attempts to solve, and why is it important? Fixing a bug? Explain the problem you're solving. Laying a foundation for future work? Explain how this fits into the bigger picture.

## What?

What does this PR change, and how does that solve the problem noted above? Call out any related changes, and add before-and-after screenshots for UI updates.

## Caveats

Are there downsides or side-effects that should be weighed against this update? Any lingering unknowns or things you can’t test without production data or traffic?

## Testing Notes

Is any special setup required to test this change? Non-obvious things that should be checked?

A list of things to test:

- [ ] Test item 1
- [ ] Test item 2
- [ ] Test item 3

## Alternatives Considered

Were there other approaches or solutions to this problem which you considered? Why were they not chosen?

## Further Reading

Were there articles or StackOverflow answers you found especially eye-opening when working on this? Slack conversation around this? Provide a link to the thread.

## Merge Instructions

Please **DO NOT** squash my commits when merging

[RA-360]: https://tractionguest.atlassian.net/browse/RA-360?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SST-20272]: https://tractionguest.atlassian.net/browse/SST-20272?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.6.2
+2.7.7
```
Comment (amaury-sie) [inline .ruby-version:1]: we're bumping ruby from 2.7.4 to 2.7.7 on guest-server. So this is to make all repos use the same ruby version

File: Gemfile.lock
Comment (amaury-sie) [inline Gemfile.lock:1]: I also ran `bundle update`, we have some vulnerabilities regarding loofah, nokogiri, etc. So I took the chance and bumped them all

File: spec/controllers/scim_rails/scim_groups_controller_spec.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@
       before { get :index }
 
       it "returns scim+json content type" do
-        expect(response.content_type).to eq "application/scim+json"
+        expect(response.media_type).to eq "application/scim+json"
```
Comment (nickrobertson) [inline spec/controllers/scim_rails/scim_groups_controller_spec.rb:14]: What caused this change?

**Review by nickrobertson — DISMISSED** [review]

File: spec/controllers/scim_rails/scim_groups_controller_spec.rb (hunk 2 above)
Comment (amaury-sie) [reply spec/controllers/scim_rails/scim_groups_controller_spec.rb:14]: It's a deprecation warning. If I'm not wrong, it's this one here:
`DEPRECATION WARNING: Rails 6.1 will return Content-Type header without modification. If you want just the MIME type, please use '#media_type' instead.`

**Review by oleg4sis — DISMISSED** [review]

**Review by oleg4sis — APPROVED** [review]

**Review by diegotoral — APPROVED** [review]

**Pull request merged**